### ISSUE TITLE
Existing log file can cause panic

### ISSUE THREAD

**Issue by wizzardx** [BourgondAries/file-rotate#20]
If you have an existing log file, and then write to a FileRotate<AppendCount> instance on it, that can cause a panic due to attempted integer wraparound.

This is specifically when the existing log file is larger than the ContentLimit::Bytes setting.

At that case, specifically in lib.rs of file_rotate create, over in fn write, this calculation in debug mode:

let bytes_left = bytes - self.count;

Can attempt to calculate a negative value, and then in debug mode, the error there is because of an attempted wraparound.

I initially picked up this error in release mode, when the panic I was receiving was because of an attempt to perform out of bounds array access, based on the wrapped-around contents of bytes_left.

Basically, it looks like there are cases where "self.count" can be larger than "bytes".

**Comment by Ploppz** [comment]
Sorry for the delay in replying. A test or minimal example that triggers this error would help. Seems important to fix.

**Comment by kotofond** [comment]
Can confirm the bug.
The following code reproduces it:
```
use std::io::Write;
use file_rotate::{ContentLimit, FileRotate};
use file_rotate::compression::Compression;
use file_rotate::suffix::AppendCount;

fn main() {
    // write 9 bytes of data
    {
        let mut file = FileRotate::new(
            "log.txt",
            AppendCount::new(2),
            ContentLimit::None,
            Compression::None,
            #[cfg(target_family = "unix")] None,
        );

        writeln!(file, "some data");
    }

    // set content limit to less than the existing file size
    {
        let mut file = FileRotate::new(
            "log.txt",
            AppendCount::new(2),
            ContentLimit::Bytes(8),
            Compression::None,
            #[cfg(target_family = "unix")] None,
        );

        writeln!(file, "some more data");
    }
}
```

panics on the second `writeln!` with
`thread 'main' panicked at 'attempt to subtract with overflow', /home/kotofond/.cargo/registry/src/github.com-1ecc6299db9ec823/file-rotate-0.7.0/src/lib.rs:643:38` message

**Comment by Ploppz** [comment]
Thanks for the example! Fixed, will release soon.
